app/lib/common/providers/chat_providers.dart

```diff
@@ -18,13 +17,20 @@ final latestMessageProvider =
   return LatestMsgNotifier(ref, convo);
 });
 
-/// Provider the profile data of a the given space, keeps up to date with underlying client
-final chatsProvider =
+/// Provider for fetching rooms list. This'll always bring up unsorted list.
+final _convosProvider =
     StateNotifierProvider<ChatRoomsListNotifier, List<Convo>>((ref) {
   final client = ref.watch(alwaysClientProvider);
   return ChatRoomsListNotifier(ref: ref, client: client);
 });
 
+/// Provider that sorts up list based on latest timestamp from [_convosProvider].
+final chatsProvider = StateProvider<List<Convo>>((ref) {
+  final convos = ref.watch(_convosProvider);
+  convos.sort((a, b) => b.latestMessageTs().compareTo(a.latestMessageTs()));
+  return convos;
+});
```

Comment:
are we sure this isn't mutating the original list? I am never 100% sure about that identity in dart...
```suggestion
final chatsProvider = StateProvider<List<Convo>>((ref) {
  final convos = List.of(ref.watch(_convosProvider));
  convos.sort((a, b) => b.latestMessageTs().compareTo(a.latestMessageTs()));
  return convos;
});
```

Reply:
Yeah it kinda does, they are mutable by default, sorry will fix this!